.github/actions/release-checklist/index.js

```diff
@@ -0,0 +1,208 @@
+const fs = require('fs');
+const core = require('@actions/core');
+const github = require('@actions/github');
+
+const commentHeader = "<!-- ubuntu-pro-client-release-checklists -->";
+const commentSectionHeader = (name) => `<!-- ubuntu-pro-client-release-checklists-${name}-header -->`;
+const commentSectionFooter = (name) => `<!-- ubuntu-pro-client-release-checklists-${name}-footer -->`;
+
+function getCommentSection(name, body) {
+    const header = commentSectionHeader(name);
+    const footer = commentSectionFooter(name);
+    try {
+        return body.split(header)[1].split(footer)[0];
+    } catch {
+        console.warn(`Failed to get "${name}" section of comment`);
+        return "";
+    }
+}
+
+
+const BUG_REFS = "bug-refs";
+function bugRefsVerified(existingSection) {
+    return existingSection.includes("- [x]")
+}
+function createBugRefsSection({
+    prTitle,
+    commits,
+    existingBody,
+}) {
+    const header = commentSectionHeader(BUG_REFS);
+    const footer = commentSectionFooter(BUG_REFS);
+    const existingSection = existingBody ? getCommentSection(BUG_REFS, existingBody) : null;
+    const verified = existingSection ? bugRefsVerified(existingSection) : false;
+    
+    
+    let lpBugs = [];
+    let ghIssues = [];
+    commits.forEach(commit => {
+        const message = commit.commit.message.toLocaleUpperCase();
+        lpBugs = lpBugs.concat(Array.from(message.matchAll(/LP: #(\d+)/g)).map(m => m[1]));
+        ghIssues = ghIssues.concat(Array.from(message.matchAll(/FIXES: #(\d+)/g)).map(m => m[1]));
+        ghIssues = ghIssues.concat(Array.from(message.matchAll(/CLOSES: #(\d+)/g)).map(m => m[1]));
+    });
+    const changelog = fs.readFileSync("./debian/changelog", { encoding: "utf8" });
+    const changelogEntries = changelog.split(/^ubuntu-advantage-tools \(/);
+    const newEntry = changelogEntries[1];
```

Comment:
Just a quick question, why are we checking the changelog here ?

Reply:
This is checking to see if all bugs that were referenced in commit messages are also references in the latest changelog entry (and this check is for release PRs only).

That's how [this comment](https://github.com/canonical/ubuntu-pro-client/pull/3104#issuecomment-2104766317) is generated

Reply:
We already had this functionality in a previous release action but it was harder to use. It originated from a recurring problem of forgetting to include bug refs in the changelog in past releases.